pinot-segment-local/src/main/java/org/apache/pinot/segment/local/utils/TableConfigUtils.java

```diff
@@ -153,10 +153,13 @@ private static Set<ValidationType> parseTypesToSkipString(@Nullable String types
    *   <li>Table name shouldn't contain dot or space in it</li>
    * </ul>
    */
-  public static void validateTableName(TableConfig tableConfig) {
+  public static void validateTableName(TableConfig tableConfig, boolean allowDots) {
     String tableName = tableConfig.getTableName();
-    if (tableName.contains(".") || tableName.contains(" ")) {
-      throw new IllegalStateException("Table name: '" + tableName + "' containing '.' or space is not allowed");
+    if (tableName.contains(".") && !allowDots) {
```

Comment:
Let's check that the table name contains at most one dot, and put some comments explaining we allow table name with embedded database name

Reply:
Added logic and also unit tests